_source/_posts/2023-07-31-workflows-workshop.md

```diff
@@ -0,0 +1,939 @@
+```
+
+The users are 
+- User: _trinity@whiterabbit.fake_, password: _Zion_
+- User: _bob@tables.fake_, password: _correct horse battery staple_
+
+
+
+### Adding an organziation to the application
+
+In a later step, you will enhance the application with a new API service to get all the todo items. The new API retrieves all the todo items belonging to an organization. 
+
+In this step, you will set up an organization in the application database. 
+
+To launch the database viewer: 
+
+1. In a new terminal tab or window, run the following command from the application folder:
+
+```
+npx prisma studio
+```
+
+You will see the following message
+
+```
+Prisma Studio is up on http://localhost:5555
+```
+
+In a new browser window, go to http://localhost:5555 to view the database. The screenshot below shows the application database. The database has two users. 
+
+{% img blog/workflows-workshop/Workflows_prisma_db.jpg alt:"Application database" %}{: .center-image }
+
+To create a new organiziation in the application database: 
+
+1. Click on the **Org** model
+2. Click **Add record** to add a new record
+3. In the **domain** column, enter **matrix.fake**
+4. In the **apikey** column, enter **131313** (you might need to scroll to the right to see this column)
+5. In the **User** column, click on **0 User** and select both users to add to this organziation (click outside the list to close it)
+5. Click **Save 1 change** button to save the new organization and its API key
+
+The organziation record should look like this (some columns are filtered/hidden to show domain, API, and User columns)
+
+{% img blog/workflows-workshop/Workflows_prisma_db_org_record.jpg alt:"Database with new organization" %}{: .center-image }
+
+
+You are ready to start the application and create the todo items. 
+
+### Running the application
+
+To start the Todo application, run the following command from the application folder: 
+
+```
+npm start
+```
+
+The create serveral todo items:
+
+1. In a browser, go to http://localhost:3000
+2. Sign in to the application. Uuse one of the following users:
+    - User: _trinity@whiterabbit.fake_, password: _Zion_
+    - User: _bob@tables.fake_, password: _correct horse battery staple_
+3. Enter several todo items
+
+{% img blog/workflows-workshop/Workflows_todoapp.jpg alt:"Todo application with several items" %}{: .center-image }
+
+You can also use Prisma Studio to view the todo items in the database. 
+
+
+## Enhancing the application with a new API
+
+In this step, you will add an API service to get all the items belonging to an organization. You will also add authentiation to the API service. 
+
+The automation you will create later in this workshop will use this API service to retrieve the todo items and generate a report. 
+
+1. In the application folder, open the file `apps/api/src/main.ts` in your favorite editor or IDE
+
+### Importing the library for API authentication
+
+1. Go to line 7, add the following code:
```

Comment:
It's dangerous to have LOC since things can change in the base app (although they shouldn't). You can give approx LOC or say "imports section at the top of the file" etc.

Reply:
Corrected.